File: Decision Tree Regression/decision_tree_regression.py
# Decision tree regression

# Import libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Import dataset
dataset = pd.read_csv('Position_Salaries.csv')

X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Fit regressor
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state = 0)
regressor.fit(X, y)

# Predict new result
y_pred = regressor.predict(6.5)

# Visualize model
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = "red")
plt.plot(X_grid, regressor.predict(X_grid), color = "blue")
plt.title("Decision tree regression model fit")
plt.xlabel("Position index")
plt.ylabel("Salary")
plt.show()

File: Decision Tree Regression/decision_tree_regression.R
# Decision tree regression

# Importing the dataset
dataset <- read.csv('Position_Salaries.csv')
dataset <- dataset[2:3]

# Decision tree regression model
library(rpart)
regressor = rpart(Salary ~ Level, data = dataset)

# Visualising the decision tree regression results (for higher resolution and smoother curve)
library(ggplot2)
x_grid <- seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue') +
  ggtitle('Decision tree regression model') +
  xlab('Level') +
  ylab('Salary')

# SVR prediction
y_pred <- predict(regressor, data.frame(Level = 6.5))